File: SurvivalPrototype/Assets/Scripts/MapGen/MapTile.js
﻿#pragma strict

import System.Collections.Generic;

public enum Dir {
	North = 0x1,
	South = 0x2,
	East = 0x4,
	West = 0x8
}

public var northDoor : GameObject;
public var southDoor : GameObject;
public var eastDoor : GameObject;
public var westDoor : GameObject;

private var openDoors : Dir = 0;

function Start() {
	// var v : int = 0;
	// for (var i : int = 0; i < 4; i++) {
	// 	v <<= 1;
	// 	v |= (Random.value > 0.5 ? 1 : 0);
	// }
	// SetDoors(v);
}

function OpenDoors(doors : Dir) {
	SetDoors(openDoors | doors);
}

function CloseDoors(doors : Dir) {
	SetDoors(openDoors & (~doors));
}

function SetDoors(doors : Dir) {
	if (openDoors != doors) {
		openDoors = doors;
		UpdateDoors();
	}
}

function UpdateDoors() {
	northDoor.SetActive(!(openDoors & Dir.North));
	southDoor.SetActive(!(openDoors & Dir.South));
	eastDoor.SetActive(!(openDoors & Dir.East));
	westDoor.SetActive(!(openDoors & Dir.West));
}

File: SurvivalPrototype/Assets/Scripts/MapGen/MapTile.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public enum Dir {
	North = 0x1,
	South = 0x2,
	East = 0x4,
	West = 0x8
}

public class MapTile : MonoBehaviour {

	public GameObject northDoor;
	public GameObject southDoor;
	public GameObject eastDoor;
	public GameObject westDoor;

	private Dir openDoors = 0;

	public void OpenDoors(Dir doors) {
		SetDoors(openDoors | doors);
	}

	public void CloseDoors(Dir doors) {
		SetDoors(openDoors & (~doors));
	}

	public void SetDoors(Dir doors) {
		if (openDoors != doors) {
			openDoors = doors;
			UpdateDoors();
		}
	}

	public Dir GetDoors() {
		return openDoors;
	}

	private void UpdateDoors() {
		northDoor.SetActive((openDoors & Dir.North) == 0);
		southDoor.SetActive((openDoors & Dir.South) == 0);
		eastDoor.SetActive((openDoors & Dir.East) == 0);
		westDoor.SetActive((openDoors & Dir.West) == 0);
	}
}
